### PR TITLE
Registration model cleanups

### PR DESCRIPTION
**Pull request by timhabermaas**
Cleaned up some things. I hope I didn't break anything.

I'm not sure what to make out of the `@birthday` instance variable. I feel like there are some more chances for yak shaving.

### CODE REVIEW

File: WcaOnRails/app/models/registration.rb [hunk 1]
```diff
@@ -25,21 +27,11 @@ def pending?
...
   end
 
-  attr_writer :birthday
-  def birthday
-    birthYear == 0 || birthMonth == 0 || birthDay == 0 ? nil : Date.parse("%04i-%02i-%02i" % [ birthYear, birthMonth, birthDay ])
-  end
-
-  before_validation :unpack_dates
   private def unpack_dates
-    if @birthday.nil? && !birthday.blank?
```
Comment (jfly) [inline WcaOnRails/app/models/registration.rb]: Removing this block of code worries me. If someone is saving a Registration without setting the `@birthday` attribute, it looks like we'll go into line 36 and clear the `birthYear`, `birthMonth`, and `birthDay` attributes. Do tests still pass with this removed?

File: WcaOnRails/app/models/registration.rb [hunk 2]
```diff
@@ -9,15 +15,11 @@ def events
...
 
-  def accepted?
-    status == "a"
-  end
-
-  def pending?
-    status == "p"
+  attr_writer :birthday
+  def birthday
+    birthYear == 0 || birthMonth == 0 || birthDay == 0 ? nil : Date.new(birthYear, birthMonth, birthDay)
```
Comment (jfly) [inline WcaOnRails/app/models/registration.rb]: Wow, that's much better, thank you.

**Comment by jfly** [comment]
Thanks for taking the initiative to clean this up. I added a comment about the `@birthday` instance variable, let me know what you think.

File: WcaOnRails/app/models/registration.rb (hunk 1 above)
Comment (timhabermaas) [inline WcaOnRails/app/models/registration.rb]: Ah, crap. I read it as `@birthday.nil? && !@birthday.blank?` which always evaluates to false. That's why I removed it. Nice catch! I'll revert the change.

Comment (jfly) [inline WcaOnRails/app/models/registration.rb]: You don't have to add one, but would our tests have caught that?

Comment (timhabermaas) [inline WcaOnRails/app/models/registration.rb]: It looks like they wouldn't. Everything's green :/.

**Comment by timhabermaas** [comment]
@jfly I removed the bad commit. Ready to merge I guess?

File: WcaOnRails/app/models/registration.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/models/registration.rb]: Ah, that's right. I wrote tests for this paradigm over in the user.rb
model, but didn't port them over to the Preregs model specs.

**Comment by jfly** [comment]
Travis is happy, so I am!

**Pull request merged**